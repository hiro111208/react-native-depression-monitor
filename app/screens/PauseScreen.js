import React from 'react';
import { 
    Image, 
    Text, 
    SafeAreaView, 
    StyleSheet, 
    TouchableOpacity, 
    Dimensions
} from 'react-native';

// Constant measurements depending on device screen size
const winWidth = Dimensions.get("window").width;
const winHeight = Dimensions.get("window").height;

const messages = 
[
    "Well done! Remember to come back and water your plant.",
    "Well done! Return to the session when you are ready.",
    "Look forward to seeing you after your break. Well done!",
    "Pause. Breathe. Keep going.\nYou can do it!",
];

/* Generate a random message from the messages array */
const randomMessage = (
     messages[Math.floor(Math.random()*messages.length)]
);

export default function PauseScreen({ props, navigation }) {
  
    return (
        <SafeAreaView style={styles.container}>
            <SafeAreaView style={[styles.center, styles.area]}>
                <SafeAreaView style={styles.messageArea}>
                    <Text style={[styles.text, styles.message]}>{ randomMessage }</Text>
                </SafeAreaView>
                <SafeAreaView style={styles.centering}>
                    <Image
                            style={styles.image}
                            resizeMode="contain"
                            source={require('../assets/pause.png')}
                    />
                </SafeAreaView>

                <SafeAreaView style={{ height: "7%" }}></SafeAreaView>

                <SafeAreaView style={styles.centering, styles.buttonArea}>
                    <TouchableOpacity
                        style={[styles.backButton, styles.shadowEffect]}
                        onPress={() => navigation.navigate("TestScreen") }
                    >
                        <Text style={[styles.text, styles.backText]}>Back to the session</Text>
                    </TouchableOpacity>

                    <TouchableOpacity
                        style={[styles.remindButton]}
                        onPress={() => navigation.navigate("TherapyScreen") } //record the time
                    >
                    </TouchableOpacity>
                </SafeAreaView>
            </SafeAreaView>
        </SafeAreaView>
    )
}

const styles = StyleSheet.create( {
    container: {
        flex: 1,
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#fff",
    },
    center: {
        height: "100%",
        width: "100%",
        alignItems: "center",
    },
    
    text: {
        fontWeight: "bold",
        textAlign: "center",
        color: "#4a4b4b",
        maxWidth: winWidth/1.3,
        lineHeight: winHeight/27,
        letterSpacing: winWidth/400,
    },
    messageArea: {
        width: "85%",
        height: "18%",
        marginTop: winHeight/15,
        borderRadius: 20,
        backgroundColor: "#fed8b1",
        alignItems: "center",
        justifyContent: "center"
    },
    message: {
        fontSize: 21,
    },
    image: {
        marginTop: winHeight/20,
        maxHeight: winHeight/2.5,
    },
    shadowEffect: {
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.2,
        shadowRadius: 3,
        elevation: 5,
        marginVertical: 5,
    },
    buttonArea: {
        width: "100%",
        height: "100%",
        alignItems: "center",    
    },
    backButton: {  
        height: "9%",
        width: "70%",
        backgroundColor: "#ffeed2",
        padding: 10,
        borderRadius: 20,
        justifyContent: "center",
    },
    backText: {
        fontSize: 18,
        color: "dimgray"
    },
    centering: {
        alignItems: "center",
        justifyContent: "center",
    },
});
