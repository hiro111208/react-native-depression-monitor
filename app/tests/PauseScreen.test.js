import React from 'react';
import renderer from 'react-test-renderer';
import PauseScreen from '../screens/PauseScreen';

test('PauseScreen renders correctly', () => {
    const tree = renderer.create(<PauseScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});