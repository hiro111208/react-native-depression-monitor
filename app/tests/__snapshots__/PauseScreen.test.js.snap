// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff7ed",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fed8b1",
          "borderRadius": 20,
          "height": "18%",
          "justifyContent": "center",
          "marginTop": 88.93333333333334,
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#4a4b4b",
              "fontWeight": "bold",
              "letterSpacing": 1.875,
              "lineHeight": 49.407407407407405,
              "maxWidth": 576.9230769230769,
              "textAlign": "center",
            },
            Object {
              "fontSize": 21,
            },
          ]
        }
      >
        Well done! Return to the session when you are ready.
         
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 30,
            }
          }
        >
          ðŸŒ¿
        </Text>
         
      </Text>
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/pause.png",
          }
        }
        style={
          Object {
            "marginTop": 66.7,
            "maxHeight": 533.6,
          }
        }
      />
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "height": "7%",
        }
      }
    />
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fecc91",
            "borderRadius": 20,
            "elevation": 5,
            "height": "9%",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 3,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#4a4b4b",
                "fontWeight": "bold",
                "letterSpacing": 1.875,
                "lineHeight": 49.407407407407405,
                "maxWidth": 576.9230769230769,
                "textAlign": "center",
              },
              Object {
                "color": "dimgray",
                "fontSize": 18,
              },
            ]
          }
        >
          Back to the session
        </Text>
      </View>
    </RCTSafeAreaView>
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;
